packages/dds/sequence/src/test/sharedString.spec.ts

```diff
@@ -318,6 +319,47 @@ describe("SharedString", () => {
             sharedString2.connect(services2);
         });
 
+        it("interval consistency", () => {
+            const collection1 = sharedString.getIntervalCollection("test");
+            sharedString.insertText(0, "xyz");
+            containerRuntimeFactory.processAllMessages();
+            const collection2 = sharedString2.getIntervalCollection("test");
+            assert.notStrictEqual(collection2, undefined, "undefined");
+            assert.strictEqual(sharedString.getText(), sharedString2.getText(), "not equal text");
+
+            sharedString.insertText(0, "abc");
+            const interval1 = collection1.add(1, 1, IntervalType.SlideOnRemove);
+            sharedString2.insertText(0, "wha");
+
+            containerRuntimeFactory.processAllMessages();
+            assert.strictEqual(sharedString.getText(), "whaabcxyz", "different text 1");
+            assert.strictEqual(sharedString.getText(), "whaabcxyz", "different text 2");
+
+            for (const interval of collection1) {
```

Comment:
Check for the interval you are looking for rather than allowing any number of same valued ones.